### DISCUSSION TITLE
Navigation does not always fully right justify

### DISCUSSION THREAD

**Discussion by jcalfee** [tailwindlabs/tailwindcss#12000] [Help] [answered]
What can cause a div to not fully right justify as I horizontally shrink the window?  It jumps as if there is some media query.  I'm having a hard time finding any media setting or CSS property in the inspector that is causing this:

Right at 640px the moon is right justified which is desired:
![image](https://github.com/tailwindlabs/tailwindcss/assets/204121/6e195abb-c2d2-4a75-8f24-3fffdb3e5155)

For a block beyond 640 it does not fully right justify (up to 767px):

![image](https://github.com/tailwindlabs/tailwindcss/assets/204121/c9a1cfc8-2562-44ff-9777-36fc5ef0d624)

Then somewhere from that point up to 768px:
![image](https://github.com/tailwindlabs/tailwindcss/assets/204121/394eda10-a564-40ac-90a1-637581e559ac)

It is just the nav with the list in it.  The main block below it is fine.
![image](https://github.com/tailwindlabs/tailwindcss/assets/204121/3429b8df-600b-49c6-b4d4-2462339f4228)

When I "Inspect" the `nav` element, the browser shows that the spacer is after the `nav` element (not the `ul` or `li`) and that `ml-auto` is working just fine (it is right justified) but the space does not show up as any kind of padding or margin.

![image](https://github.com/tailwindlabs/tailwindcss/assets/204121/29725bea-7c11-49fc-bb42-cb1811e7f8a6)

I don't see anything when I filter css properties like "media".  It must have something to do with media, I just don't know why it is picking on this first element and not the `main` block and I'm not sure where to find this setting in the inspector.

```tsx
          <nav class="bg-sky-800 flex">
            <ul class="container flex p-3 text-gray-200">
              <li class={`border-b-2 inline-flex ${active("/")} mx-1.5 sm:mx-6`}>
                <A href="/">Home</A>
              </li>
              <li class={`border-b-2 ${active("/about")} mx-1.5 sm:mx-6`}>
                <A href="/about">About</A>
              </li>
              {/* <li class="flex-grow"></li> */}
              <li class="ml-auto pt-1">
                <ThemeButton class="" />
              </li>
            </ul>
          </nav>
```

Thank you!

**Comment by wongjn** [comment, marked as answer]
The behavior you are seeing is due to the `max-width` properties from the `container` class being applied per breakpoint. Check [the documentation on `container`](https://tailwindcss.com/docs/container#class-reference) for details.

**Reply by jcalfee** [reply]
`container max-w-full` did the trick!  thank you
